src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/AbstractGenerateNuGetPackageCommand.cs

```diff
@@ -93,12 +93,14 @@ protected override async Task<bool> TryHandleCommandAsync(IProjectTree node, boo
                 var projectVsHierarchy = (IVsHierarchy)Project.Services.HostObject;
                 ErrorHandler.ThrowOnFailure(_buildManager!.SaveDocumentsBeforeBuild(projectVsHierarchy, (uint)VSConstants.VSITEMID.Root, 0 /*docCookie*/));
 
-                // Enable generating package on build ("GeneratePackageOnBuild") for all projects being built.
-                _generatePackageOnBuildPropertyProvider.OverrideGeneratePackageOnBuild(true);
+                // Turn off "GeneratePackageOnBuild" because otherwise the Pack target will not do a build, even if there is no built output
+                _generatePackageOnBuildPropertyProvider.OverrideGeneratePackageOnBuild(false);
 
-                // Kick off the build.
                 uint dwFlags = (uint)(VSSOLNBUILDUPDATEFLAGS.SBF_SUPPRESS_SAVEBEFOREBUILD_QUERY | VSSOLNBUILDUPDATEFLAGS.SBF_OPERATION_BUILD);
-                ErrorHandler.ThrowOnFailure(_buildManager.StartSimpleUpdateProjectConfiguration(projectVsHierarchy, null, null, dwFlags, 0, 0));
+
+                // We tell the Solution Build Manager to Package, which will call the Pack target, which will build if necessary.
+                uint[] buildFlags = new[] { VSConstants.VS_BUILDABLEPROJECTCFGOPTS_PACKAGE };
+                ErrorHandler.ThrowOnFailure(_buildManager.StartUpdateSpecificProjectConfigurations(1, new[] { projectVsHierarchy }, null, null, buildFlags, null, dwFlags, 0));
             }
```

Comment:
Is there an entry-point that "package" (this is solution "publish", right?) can be executed elsewhere via a command or menu item? If so, this code won't be running and we won't override the global property.

Reply:
This is that command and menu item. There isn't anything in VS itself that I could find that uses the package option

Reply:
Can you create an AppXPackage producing project and reference this project and try publishing that? I'm pretty sure that will skip this global property registering.

Reply:
The way I found this was searched for all usages of VS_BUILDABLEPROJECTCFGOPTS_PACKAGE.

Reply:
Yeah, I searched for the same thing and also found the AppX packager, which is also linked in the AzDo issue. Maybe I'm missing something, but I don't see how it's related? It only calls the Package target on the first project in the selection, not on dependencies (see [this comment](https://dev.azure.com/devdiv/DevDiv/_git/VS?path=%2Fsrc%2Fvsproject%2FPackageAndDeploy%2FPackagingService%2FCommands%2FPackaging.cs&version=GBmaster&line=621&lineStyle=plain&lineEnd=622&lineStartColumn=1&lineEndColumn=1)). Even if it did do it on dependencies, it wouldn't work today because we don't currently supply a value for PackageAction, so this change only improves things. And finally, if you were creating a package for a project, you wouldn't want individual packages for dependant projects anyway, you'd want the dependencies included in your package.